Source/ProgressionSystemRuntime/Private/Components/PSSpotComponent.cpp

```diff
@@ -67,16 +76,45 @@ void UPSSpotComponent::OnUnregister()
 	Super::OnUnregister();
 }
 
-void UPSSpotComponent::OnCurrentScoreChanged_Implementation(const FPSSaveToDiskData& CurrentSaveToDiskDataRow, const FPSRowData& CurrentProgressionSettingsRow)
+// Check is player is allowed to play with current skin if not switch to allowed
+void UPSSpotComponent::TryRestorePlayerSkin()
 {
-	const APlayerCharacter* PlayerCharacter = UMyBlueprintFunctionLibrary::GetLocalPlayerCharacter();
+	// it's possible that spot might not be loaded till that time so no ensure added
+	const UPSSpotComponent* CurrentSpot = UPSWorldSubsystem::Get().GetCurrentSpot();
+	if (CurrentSpot == nullptr || CurrentSpot != this)
+	{
+		return;
+	}
 
-	if (!ensureMsgf(PlayerCharacter, TEXT("ASSERT: [%i] %hs:\n'PlayerCharacter' is not valid!"), __LINE__, __FUNCTION__))
+	UMySkeletalMeshComponent& MeshComp = GetMeshChecked();
+	const int32 CurrentSkinIndex = MeshComp.GetAppliedSkinIndex();
+	bool bIsCurrentSkinAvailable = MeshComp.IsSkinAvailable(CurrentSkinIndex);
+
+	// Current skins is available no need to switch to last avaialble
+	if (bIsCurrentSkinAvailable)
 	{
 		return;
 	}
-	const FPlayerTag& PlayerTag = PlayerCharacter->GetPlayerTag();
-	if (GetMeshChecked().GetPlayerTag() == PlayerTag)
+
+	// find last unlocked skin 
+	for (int32 Count = CurrentSkinIndex; Count >= 0; Count--)
+	{
+		bool bIsLastUnlockedFound = Count > 0 ? MeshComp.IsSkinAvailable(Count) : bIsCurrentSkinAvailable = true;
```

Comment:
1. The main problem here: assigning `= true` at the end of ternary is mistake, you probably wanted the equality `==` operator instead, but mistakenly used assigning `=`
2. Please dont forget about `const`

### Expected Result
Where I additionally further simplified the line by replacing ternary with simple condition, that is the same:
```
const bool bIsLastUnlockedFound = Count > 0 && MeshComp.IsSkinAvailable(Count) || bIsCurrentSkinAvailable;
```

Reply:
here I need to assign bIsCurrentSkinAvailable to true and then assign bIsLastUnlcokedUnlockedFound. 
I can simplify reading this to something like:

```suggestion
bool bIsLastUnlockedFound = false;

		if (Count > 0)
		{
			bIsLastUnlockedFound = MeshComp.IsSkinAvailable(Count);
		}
		else
		{
         		bIsCurrentSkinAvailable = true;
			bIsLastUnlockedFound = bIsCurrentSkinAvailable;
		}
```
Let me know if I need to incorporate those changes. 
by far const  added as part of 
https://github.com/moinm3uw/ProgressionSystem/pull/63/commits/c1a8d9d82b0be59e5c947dc44d8f756ff2aedfbb

Reply:
@moinm3uw, current logic still looks overcomplicated. Please simplify it by removing the unused `bIsCurrentSkinAvailable` variable entirely, avoid assignments inside ternary, and replace the ternary with a clean inline condition using logical operators. This will improve clarity and prevent side effects that may confuse future readers.

Next simplified logic should do all the same as original:
```
// Skip if current skin is already available
if (MeshComp.IsSkinAvailable(CurrentSkinIndex))
{
	return;
}

// Find last available skin from current index downward
for (int32 Count = CurrentSkinIndex; Count >= 0; Count--)
{
	if (Count == 0 || MeshComp.IsSkinAvailable(Count))
	{
		MeshComp.ApplySkinByIndex(Count);

		constexpr bool bApplySkin = true;
		RefreshAmountOfUnlockedSkins(bApplySkin);
		break;
	}
}
```

Reply:
done https://github.com/moinm3uw/ProgressionSystem/pull/63/commits/07bfc89aee38fa7a6c5ba3dc1bdae1a8204987da